libs/lcos_local/src/composable_guard.cpp

```diff
@@ -225,12 +225,36 @@ namespace hpx { namespace lcos { namespace local {
         }
     };
 
+    namespace detail {
+        struct empty_helper
+        {
+            static guard_task*& get_empty_guard_task()
+            {
+                static guard_task* empty = new guard_task;
+                return empty;
+            }
+
+            empty_helper() = default;
+            ~empty_helper()
+            {
+                auto& empty = get_empty_guard_task();
+                delete empty;
+                empty = nullptr;
+            }
+        };
+
+        empty_helper empty_helper_{};
```

Comment:
Why is this needed ? (still working on my c++ skills) ^^'

Reply:
This creates a global instance of `empty_helper`. We rely on the destructor of this (global) object to clean up the allocated memory.